spec/httparty/request_spec.rb

```diff
@@ -342,6 +342,25 @@
         expect(CGI.unescape(body)).to eq("foo=bar&foo=baz&page=1")
       end
     end
+
+    context "when body is multipart" do
+      it "sets header Content-Type: multipart/form-data; boundary=" do
+        @request.options[:body] = {file: File.open(File::NULL, 'r')}
+        @request.send(:setup_raw_request)
+        headers = @request.instance_variable_get(:@raw_request).each_header.to_h
+        expect(headers['content-type']).to match(%r{^multipart/form-data; boundary=---})
+      end
+
+      context "and header Content-Type is provided" do
+        it "overwrites the header to: multipart/form-data; boundary=" do
+          @request.options[:body] = {file: File.open(File::NULL, 'r')}
+          @request.options[:headers] = {'Content-Type' => 'application/x-www-form-urlencoded'}
+          @request.send(:setup_raw_request)
+          headers = @request.instance_variable_get(:@raw_request).each_header.to_h
+          expect(headers['content-type']).to match(%r{^multipart/form-data; boundary=---})
```

Comment:
This is nasty copypaste code, but unfortunately that’s how the rest of the tests in this file is written, so to be consistent…